### PR TITLE
Adds logger on cloud code request object

### PR DESCRIPTION
**Pull request by flovilmart**
@blacha that's what you wanted I guess.

### CODE REVIEW

File: src/Routers/FunctionsRouter.js [hunk 1]
```diff
@@ -38,8 +41,13 @@ export class FunctionsRouter extends PromiseRouter {
         params: params,
         master: req.auth && req.auth.isMaster,
         user: req.auth && req.auth.user,
-        installationId: req.info.installationId
+        installationId: req.info.installationId,
+        logger
       };
+      
+      request.log = function() {
```
Comment (drew-gross) [inline src/Routers/FunctionsRouter.js]: This seems redundant with `logger`. Why have both?

Comment (flovilmart) [inline src/Routers/FunctionsRouter.js]: to have a quick access to it, but I can remove.

**Comment by drew-gross** [comment]
Cool. LGTM, but lets get some opinions from @gfosco and @nlutsenko since this is a new public API.

**Comment by nlutsenko** [comment]
Lgtm. Logger as a property on a request sounds good to me. I wonder if a global would be something safer to maintain, but I guess it doesn't necessarily matter much.

**Comment by flovilmart** [comment]
@nlutsenko I prefer the scope based logger so we can later maybe add an app id to log prefixes

**Comment by drew-gross** [comment]
It could be app scoped. That would be in between request scoped and global scoped. Too bad cloud code is already global scoped :(

Or with request scoped we could add the function/trigger name. That seems like a quite useful feature actually.

**Comment by blacha** [comment]
I have never used winston before so forgive my ignorance. Is there no way to do child loggers in winston?

If you used something like bunyan inside the ``handleCloudFunction`` you could do something like 
```javascript
{ 
    ...
    log: currentLogger.child({ 
        function: req.params.functionName,
        user: req.auth && req.auth.user && req.auth.user.id
   })
```

Then every call inside of the cloud function would log out those parameters as well

```javascript
req.log.info('message')
// {fuction: 'foo', user: 'bar', level: 'info', msg: 'message'}
```

**Comment by flovilmart** [comment]
the logger is already a child logger, that will log specifically in a cloud-code file. Like discussed above, we could add more default parameters like function name, trigger type etc... you can also do it like in the tests.

**Comment by flovilmart** [comment]
@drew-gross it's request scoped and generated for each request life-cycle, we can add more parameters like functionName, trigger etc.. to it if we want

**Comment by drew-gross** [comment]
@blacha note that we are trying to avoid being tied to `winston`, we have `LoggerAdapter` so you can provide your own logger (for e.g. logging to some 3rd party service)

**Comment by flovilmart** [comment]
@drew-gross on that note, it never worked that well as at many places, the logger adapter is not available, that's why I added the ./logger. which is more 'static' but we could interface easily from there for writing logs. the loggerAdapter is more for reading.

**Comment by blacha** [comment]
@flovilmart  It isn't really a "child" logger in the ``bunyan`` sense, you have actually created a entire new logger pointing at a different file.

How does the ``DailyRotateFile`` handle having 10, 100, 1000? instances pointing at the same file?

The bunyan child logger only add parameters to log messages, and uses the parent's logging streams as the transports.

**Comment by flovilmart** [comment]
I believe you're going off topic here. There is only one file log per server, it's up to you to aggregate with your own means, with log entries or another strategy. I don't believe we'll implement a logging aggregator as parse dashboard don't support multiple instances either.

**Comment by blacha** [comment]
@flovilmart  you have made 2 ``DailyRotateFile`` onto each of ``parse-server.info`` and ``parse-server.err`` by using ``generateTransports()`` inside of ``addGroup``

I don't think its off topic, as there is already discussion about adding more information into a per request basis, cloud function name, request username. If that is going to be handled by some sort of ``addGroup`` function then in theory you could get 10, 100, 1000 ``DailyRotateFile`` loggers made.

**Comment by flovilmart** [comment]
I'm closing that PR and let you implement it then

**Pull request closed without merging**